### DISCUSSION TITLE
Question again: Can I combine different backup files (.bck)?

### DISCUSSION THREAD

**Discussion by spacex-3** [teslamate-org/teslamate#3312] [Q&A]
I have two backup files (bck) with different periods.
Can these two be restored to one teslamate?

The teslamate1.bck file that I backed up from 2021.01-2022.12 (700mb)
There is a teslamate2.bck file that backed up data from 2023.01-2023.06 (300mb) by installing teslamate again because the older docker container was broken once.

I installed a new teslamate and restored teslamate1.bck 1 bck as guided.
Restored normally.

After that, I have no idea to execute restore for another period of bck file.

So how to finish that? Thanks!

**Comment by DrMichael** [comment]
To answer the question: yes.

But with tremendous efforts: in one dataset you have to renumber the ids of all tables, to make them diffferent from the other dataset. That means, you also have to adapt the usage of the tables in all other tables. After that having done you can just combine the tables.

Thinking about it, what one could try:

- Get all the maximun id's of the tables of database A
- Then for all tables in database B
    - Add the respective maximum id number to
        - the all id's in the corresponding table
        - to all ocurrences of that's table id in the other tables

(Nice database programming exercise...:-)

If the cars were identical you could skip that for the _cars_ table.

I don't know, whether there might be other issues...

**Reply by spacex-3** [reply]
Hello Michael! I'm only a user without any database skills.

So I'm really want to understant what you mentioned above and no results.

It would be great if you can tell the tools and details to finish the work.

And I believe there must be some great guys finish that.
